src/content/resources/javascript/variables.md

```diff
@@ -0,0 +1,188 @@
+```
+
+## Dynamic Types
+Variables in Javascript do not have explicit types, as some other languages (eg: Java) do. This means that when declaring a variable, you only use `name = value`, and don't need to add a type as well. The type is automatically chosen based on the value.
+
+```js
+let myString = "Hello";
+let myBoolean = true; // false is also valid
+let myNumber = 1;
+let myFloat = 1.0;
+```
+
+## Let vs Const
+Javascript has two "types" of variable declaration: `let` and `const`. These behave in largely the same way, with **one important difference**:
+
+- Let: the value of variables using `let` can be changed throughout the course of your program.
+- Const: the value of variables using `const` cannot be changed after they are defined.
+
+```js
+let myVar = "Hello World";
+myVar = "New Value!"; // This is ok, and myVar's value will be changed.
+
+const myVar2 = "Hello World";
+myVar2 = "New Value!" // This will fail, as myVar2 has already been defined.
+```
+
+If the value of the variable will only ever have one value, `const` should be used to define it. This is safer, as if you try to re-assign it somewhere else you will get an error, instead of having unexpected behaviour by accidentally overwriting the value.
+
+If the value of the variable needs to change throughout the course of a program (user input, calculations, etc...), `let` should be used to define it.
+
+### Const Quirks:
+The `const` declaration in Javascript allows you to create a variable that cannot be redeclared after it has been declared initially. This is similar to a lot of other languages.
+
+However, in Javascript, there is an edge case to be aware of when using objects and arrays.
+
+```js
+const myArray = [1,2,3];
+console.log(myArray); // [1,2,3]
+
+myArray.push(4);
+myArray.push(5);
+console.log(myArray); // [1,2,3,4,5]
+
+const myObj = { foo: "bar" };
+console.log(myObj); // {foo: bar}
+
+myObj.foo = "re-assigned";
+console.log(myObj); // {foo: re-assigned}
+```
+
+This, however, will still not work:
+```js
+const myArray = [1,2,3];
+
+myArray = [1,2,3,4,5] // Error: myArray has already been defined
+```
+
+## What about var?
+As well as using `let` or `const` to define variables, there is a 3rd way that exists using `var`. Var is from older versions of Javascript from before let and const were introduced. Nowadays, usoing `let` or `const` is recommended over `var`.
+
+There are a number of problems that come from using `var` to declare variables, that are fixed by using `let` or `const` instead. 
+
+### Block Scoping
+Var is not "block scoped" (a block is anything that contains `{}`, so functions, if, for loop, etc...) This means that variables declared using `var` are not *just* defined in the block they are declared, but declared and can be accessible globally. This can have un-intended side effects.
+
+```js
+function myFunction() {
+  var myVar = "test" //declared with the 'var' keyword
+}
+console.log(myVar); // variable can be used outside myFunction
```

Comment:
Little error here, ``var`` is function-scoped (or global if it's not declared inside a function), so ``myVar`` isn't accessible outside of ``myFunction``

Reply:
Updated now, thanks for the spotting that :)